ui/hooks/useTransactionEventFragment.js

```diff
@@ -0,0 +1,22 @@
+import { useCallback } from 'react';
+
+import { useGasFeeContext } from '../contexts/gasFee';
+import { updateEventFragment } from '../store/actions';
+
+export const useTransactionEventFragment = () => {
```

Comment:
[Q] out of curiosity how many callsites for this do not have access to txMeta.id?

Reply:
[GasFeeContext](https://github.com/MetaMask/metamask-extension/blob/develop/ui/contexts/gasFee.js) has been added to the root components of all transaction flows - simplesend, swap, cancel/speedup, approve token. Thus this should be available to all of transaction related components.